compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/KnownTargets.cpp

```diff
@@ -215,48 +235,147 @@ std::optional<TargetDetails> getAMDGPUTargetDetails(StringRef target) {
       .Case("rx7700xt", TargetDetails{rdna3Wgp, &rx7700xtChip})
       .Cases("rdna3", "gfx1100", "gfx1101", "gfx1102", "gfx1103", "gfx1150",
              "gfx1151", TargetDetails{rdna3Wgp, nullptr})
+      .Cases("rdna2", "gfx1030", "gfx1031", "gfx1032", "gfx1033", "gfx1034",
+             "gfx1035", "gfx1036", TargetDetails{rdna2Wgp, nullptr})
+      .Cases("rdna1", "gfx1010", "gfx1011", "gfx1012", "gfx1013",
+             TargetDetails{rdna1Wgp, nullptr})
       .Default(std::nullopt);
 }
 
 StringRef normalizeAMDGPUTarget(StringRef target) {
   if (target.starts_with("gfx"))
     return target;
 
+  // We cannot accept rdnaN as a target for LLVM AMDGPU backend; so the
+  // following is only meant for Vulkan but not HIP.
+  if (target.starts_with("rdna"))
+    return target;
+
   return llvm::StringSwitch<StringRef>(target.lower())
       .Case("mi300x", "gfx942")
       .Case("mi300a", "gfx940")
       .Cases("mi250x", "mi250", "mi210", "cdna2", "gfx90a")
+      .Case("cdna1", "gfx908")
       .Cases("rx7900xtx", "rx7900xt", "gfx1100")
       .Cases("rx7800xt", "rx7700xt", "gfx1101")
       .Default(StringRef());
 }
 
+//===----------------------------------------------------------------------===//
+// Known Apple target details
+//===----------------------------------------------------------------------===//
+
+std::optional<TargetDetails> getAppleTargetDetails() {
+  ComputeBitwidths computeBitwdiths =
+      allIntComputeBits | ComputeBitwidths::FP32 | ComputeBitwidths::FP16;
+  // clang-format off
+  static const WgpDetails wgp = {
+      computeBitwdiths,   allStorageBits,     allSubgroupOps,  allDotProductOps,
+      /*mmaCount=*/0,     /*mmaOps=*/nullptr, {32, 32},
+      {1024, 1024, 1024}, 1024,               32 * 1024};
+  // clang-format on
+
+  return TargetDetails{&wgp, nullptr};
+}
+
+//===----------------------------------------------------------------------===//
+// Known ARM target details
+//===----------------------------------------------------------------------===//
+
+const WgpDetails *getValhallWgpDetails() {
+  ComputeBitwidths computeBitwdiths =
+      allIntComputeBits | ComputeBitwidths::FP32 | ComputeBitwidths::FP16;
+  // clang-format off
+  static const WgpDetails valhallWgp = {
+      computeBitwdiths,   allStorageBits,     allSubgroupOps,  allDotProductOps,
+      /*mmaCount=*/0,     /*mmaOps=*/nullptr, {16},            {512, 512, 512},
+      512,                32 * 1024};
```

Comment:
What is the motivation behind the formatting here? I see the max threads per dim on the second line here, unlike about where it's on the third line (only commenting because I see `clang-format off` and trying to understand what this is trying to capture).

Reply:
Just trying to organize this a bit better to avoid spanning many lines.. No particular reasons. You can call this formating a bit personal taste though. :)